test/OptionSettlementEngine.integration.t.sol

```diff
@@ -242,44 +242,45 @@ contract OptionSettlementIntegrationTest is BaseEngineTest {
         vm.startPrank(BOB);
     }
 
-    function test_integrationWriteExerciseAddBuckets() public {
-        vm.startPrank(ALICE);
-        uint256[7] memory claimRatios;
-        uint112 targetBuckets = 7;
-        uint256 i;
-        for (i = 0; i < targetBuckets; i++) {
-            engine.write(testOptionId, targetBuckets);
-            engine.exercise(testOptionId, 1);
-        }
-
-        // 49 written, 7 exercised
-        for (i = 1; i <= targetBuckets; i++) {
-            IOptionSettlementEngine.Claim memory claimData = engine.claim(testOptionId + i);
-            uint256 claimRatio = FixedPointMathLib.divWadDown(claimData.amountExercised, claimData.amountWritten);
-            emit log_named_uint("amount written WAD     ", claimData.amountWritten);
-            emit log_named_uint("amount exercised WAD   ", claimData.amountExercised);
-            // dividing by the amount written in the claim recovers the bucket ratio WAD
-            emit log_named_uint("claim ratio WAD        ", claimRatio);
-            claimRatios[i - 1] = claimRatio;
-        }
-
-        uint256 bucketRatio0 = FixedPointMathLib.divWadDown(3, 7);
-        uint256 bucketRatio1 = FixedPointMathLib.divWadDown(2, 7);
-        uint256 bucketRatio2 = 0;
-
-        // Claim 1 is exercised in a ratio of 3/7
-        assertEq(claimRatios[0], bucketRatio0);
-
-        // Claims 2 and 3 are exercised in a ratio of 2/7
-        assertEq(claimRatios[1], bucketRatio1);
-        assertEq(claimRatios[2], bucketRatio1);
-
-        // Claims 4, 5, 6, and 7 are not exercised (0/7)
-        assertEq(claimRatios[3], bucketRatio2);
-        assertEq(claimRatios[4], bucketRatio2);
-        assertEq(claimRatios[5], bucketRatio2);
-        assertEq(claimRatios[6], bucketRatio2);
-    }
+    // TODO resolve test which broke after removing 0s from newOptionType
```

Comment:
Can we either fix these in this PR or drop this commented code

Reply:
I looked into this failing test. It seems the different initial option key / settlement seed causes pretty significant changes to exercise assignment, I believe driven by L788 `uint96 exerciseIndex = uint96(optionRecord.settlementSeed % numUnexercisedBuckets);`. Raw bucket ratio outputs included below.

Taking a step back, what do we want to check w/ this test? What if we remove this test and write a new _unit/fuzz_ test that checks exercise assignment is fair and works as expected (notably missing from unit tests rn).

Before changing initial settlement seed (audit-fixes branch):

```
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 3000000000000000000
  claim ratio WAD        : 428571428571428571
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 2000000000000000000
  claim ratio WAD        : 285714285714285714
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 2000000000000000000
  claim ratio WAD        : 285714285714285714
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 0
  claim ratio WAD        : 0
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 0
  claim ratio WAD        : 0
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 0
  claim ratio WAD        : 0
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 0
  claim ratio WAD        : 0
```
  
After changing initial settlement seed (threat-model-tests branch):
  
```
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 4000000000000000000
  claim ratio WAD        : 571428571428571428
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 333333333333333333
  claim ratio WAD        : 47619047619047619
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 333333333333333333
  claim ratio WAD        : 47619047619047619
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 333333333333333333
  claim ratio WAD        : 47619047619047619
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 500000000000000000
  claim ratio WAD        : 71428571428571428
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 500000000000000000
  claim ratio WAD        : 71428571428571428
  amount written WAD     : 7000000000000000000
  amount exercised WAD   : 1000000000000000000
  claim ratio WAD        : 142857142857142857
```

Reply:
@neodaoist let's just fix the integration test and get this merged. Consider R&D on the core contracts frozen in the new year.

Reply:
Agree 100%. I can't figure out how to fix this test tho, could use help from you or Flip. With the new settlement seed, all claims get some exercised from them. I'm not sure how to calculate what we expect each claim's ratio to be.